extmod/modure.c

```diff
@@ -344,6 +344,10 @@ STATIC mp_obj_t re_sub_helper(size_t n_args, const mp_obj_t *args) {
                         vstr_add_strn(&vstr_return, start_match, end_match - start_match);
                     }
                 }
+                if ('\\' == *repl) {
```

Comment:
I think this should be an `else if`

Reply:
Also for coding style, should be `*repl == '\\'`

Reply:
Ah yes of course, I had that initially but removed it for some reason